frb_dart/lib/src/generalized_frb_rust_binding/_io.dart

```diff
@@ -26,8 +37,14 @@ class GeneralizedFrbRustBinding {
   }
 
   /// {@macro flutter_rust_bridge.only_for_generated_code}
-  void initFrbDartApiDl() =>
-      _binding.frb_init_frb_dart_api_dl(ffi.NativeApi.initializeApiDLData);
+  void initShutdownWatcher() {
+    _shutdownWatcher = ShutdownWatcher(_binding.frb_shutdown_callback);
```

Comment:
nit: add `assert (_shutdownWatcher == null);` as an extra sanity check

Reply:
You are right that it should be `??=`, but cant do assert here. There might be multiple frb's in scope. Like your project is using frb, and one of your dependencies also use frb. So `initShutdownWatcher` will end up being called multiple times, but for different libraries and we want `_ShutdownWatcher` to be either a global variable or a static field so there will be only one.